### PR TITLE
Managing UI color at various places

### PR DESCRIPTION
**Pull request by heysadboy**
Changing edge glow color of recyclerviews of tracks and sessions speaker activity

Fixes https://github.com/fossasia/open-event-android/issues/1616

Screenshots for the change: 

<table>
<td>

</td>
<td>

</td>
</table>

### CODE REVIEW

**Comment by iamareebjamal** [comment]
@heysadboy Please only make changes which are required

Here, formatting has completely changed the order of the old file. Please revert and only make the changes which are required, don't format and rearrange entries

File: android/app/src/main/java/org/fossasia/openevent/utils/Views.java [hunk 1]
```diff
@@ -113,6 +115,31 @@ public static void changeGlowColor(int color, NestedScrollView scrollView) {
...
+            }
+            for (final String name : new String[]{"mTopGlow", "mBottomGlow", "mLeftGlow", "mRightGlow"}) {
+                final Field field = clazz.getDeclaredField(name);
+                field.setAccessible(true);
+                final Object edge = field.get(recyclerView);
+                final Field fEdgeEffect = edge.getClass().getDeclaredField("mEdgeEffect");
+                fEdgeEffect.setAccessible(true);
+                ((EdgeEffect) fEdgeEffect.get(edge)).setColor(color);
+            }
+        } catch (final Exception ignored) {
```
Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/utils/Views.java]: Don't use try catch 

Use build versions to check for lollipop or above

Comment (heysadboy) [inline android/app/src/main/java/org/fossasia/openevent/utils/Views.java]: @iamareebjamal I am using `@TargetApi(LOLLIPOP)` above the function. Try and catch is for the error thrown by functions. I am not sure what you are trying to say.

Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/utils/Views.java]: OK then, but please catch only specific exceptions and if ignored, write comments in each block

Comment (heysadboy) [inline android/app/src/main/java/org/fossasia/openevent/utils/Views.java]: Comments about the errors in catch clause?

Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/utils/Views.java]: like this 

````Java
try {
    ....
} catch (XYZException xzy) {
    // No action to be taken
}
````

**Comment by heysadboy** [comment]
@iamareebjamal  I have made the changes, please see.

File: android/app/src/main/java/org/fossasia/openevent/activities/SessionDetailActivity.java [hunk 2]
```diff
@@ -124,7 +124,7 @@ public void onCreate(final Bundle savedInstanceState) {
         disposable = new CompositeDisposable();
 
         setSupportActionBar(toolbar);
-        if(getSupportActionBar() != null) getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+        if (getSupportActionBar() != null) getSupportActionBar().setDisplayHomeAsUpEnabled(true);
```
Comment (arp95) [inline android/app/src/main/java/org/fossasia/openevent/activities/SessionDetailActivity.java]: please remove the space after if. do that at all places please.

File: android/app/src/main/java/org/fossasia/openevent/activities/SessionDetailActivity.java [hunk 3]
```diff
@@ -165,7 +165,7 @@ public void onCreate(final Bundle savedInstanceState) {
         speakersRecyclerView.setItemAnimator(new DefaultItemAnimator());
 
         int color = TrackColors.getColor(trackId);
-        if(trackId == -1 || color == -1) {
+        if (trackId == -1 || color == -1) {
             disposable.add(dbSingleton.getTrackByNameObservable(trackName)
```
Comment (arp95) [inline android/app/src/main/java/org/fossasia/openevent/activities/SessionDetailActivity.java]: please ensure the codebase isnt affected from your changes in future.

File: android/app/src/main/java/org/fossasia/openevent/activities/TrackSessionsActivity.java [hunk 4]
```diff
@@ -87,13 +87,13 @@ protected void onResume() {
         //setting the grid layout to cut-off white space in tablet view
         DisplayMetrics displayMetrics = this.getResources().getDisplayMetrics();
         float width = displayMetrics.widthPixels / displayMetrics.density;
-        int spanCount = (int) (width/250.00);
+        int spanCount = (int) (width / 250.00);
```
Comment (arp95) [inline android/app/src/main/java/org/fossasia/openevent/activities/TrackSessionsActivity.java]: please maintain the codebase.

File: android/app/src/main/java/org/fossasia/openevent/activities/SessionDetailActivity.java [hunk 5]
```diff
@@ -271,10 +271,18 @@ private void setUiColor(int color) {
...
             getWindow().setStatusBarColor(darkColor);
+            Views.setEdgeGlowColorScrollView(color, scrollView);
+            speakersRecyclerView.addOnScrollListener(new RecyclerView.OnScrollListener() {
+                @Override
+                public void onScrollStateChanged(RecyclerView recyclerView, int newState) {
+                    super.onScrollStateChanged(recyclerView, newState);
+                    Views.setEdgeGlowColorRecyclerView(speakersRecyclerView, color);
+                }
+            });
+        }
```
Comment (arp95) [inline android/app/src/main/java/org/fossasia/openevent/activities/SessionDetailActivity.java]: does it seem necessary here? Over tracks session activity it looks great but over here didnt look good to me. let others review.

File: android/app/src/main/java/org/fossasia/openevent/activities/TrackSessionsActivity.java (hunk 4 above)
Comment (heysadboy) [inline android/app/src/main/java/org/fossasia/openevent/activities/TrackSessionsActivity.java]: @arp95 This happens when I auto indent and putting spaces between the operatorand operand is encouraged. Read more about it here https://google.github.io/styleguide/javaguide.html . Do you also see other places where there is space already?

File: android/app/src/main/java/org/fossasia/openevent/activities/SessionDetailActivity.java (hunk 5 above)
Comment (heysadboy) [inline android/app/src/main/java/org/fossasia/openevent/activities/SessionDetailActivity.java]: All the edge glow were of particular colour and I don't think we should leave it red anywhere else. Please see the screenshots above. @the-dagger What's your opinion?

File: android/app/src/main/java/org/fossasia/openevent/activities/TrackSessionsActivity.java (hunk 4 above)
Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/activities/TrackSessionsActivity.java]: It is encouraged but makes unnecessary changes in PR. Which is not good. A separate issue handling code reformat can be made

Comment (heysadboy) [inline android/app/src/main/java/org/fossasia/openevent/activities/TrackSessionsActivity.java]: @iamareebjamal Thanks for clarifying, I will remember not to to this from next time.

File: android/app/src/main/java/org/fossasia/openevent/activities/SessionDetailActivity.java (hunk 5 above)
Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/activities/SessionDetailActivity.java]: Yeah @arp95, we should remove the fossasia specific artifacts and make the app as neutral as possible

**Comment by heysadboy** [comment]
@the-dagger Please see.
